### PR TITLE
Missing bounds check for offsetByCodePoints

### PR DESCRIPTION
**Pull request by mikezhang1234567890**
Add missing bounds check for offsetByCodePoints method in String, StringBuffer, and StringBuilder classes.
Add tests for offsetByCodePoints which result in out of bounds indices. 
Change a StringBuffer constructor to StringBuilder in StringBuilder test.

Fixes https://github.com/eclipse-openj9/openj9/issues/16188

Testing done on internal jenkins:
[Compile](https://hyc-runtimes-jenkins.swg-devops.com/job/Pipeline-Build-Test-Personal/14589/) - only difference between the commit used for the build and this PR is a copyright update.

`sanity.functional` tests running with `-XX:+CompactStrings`:
[JDK8](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/29008/)
[JDK11](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/29009/)

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/StringBuffer.java [hunk 1]
```diff
@@ -2759,7 +2759,7 @@ public synchronized int offsetByCodePoints(int start, int codePointCount) {
 		if (String.COMPACT_STRINGS && count >= 0) {
 			int index = start + codePointCount;
 
-			if (index >= currentLength) {
+			if ((index >= currentLength) || (index < 0)) {
```
Comment (keithc-ca) [inline jcl/src/java.base/share/classes/java/lang/StringBuffer.java:2762]: This is not consistent with `String` - I think `>=` should be `>`; likewise in `StringBuilder`.
It would be good to have tests for the boundary conditions.

**Comment by mikezhang1234567890** [comment]
Pushed new commit to fix test errors. Will update the test links.

File: test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_StringBuilder.java [hunk 2]
```diff
@@ -265,6 +265,49 @@ public void test_codePointCount() {
 		AssertJUnit.assertTrue("wrong count 2", sb.codePointCount(1, 4) == 2);
 	}
 
+	/**
+	 * @tests java.lang.StringBuilder#offsetByCodePoints(int, int)
+	 */
+	@Test
+	public void test_offsetByCodePoints() {
+		StringBuilder s1 = new StringBuilder("A\ud800\udc00C");
+		AssertJUnit.assertTrue("wrong offset 0, 0", s1.offsetByCodePoints(0, 0) == 0);
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/java/lang/Test_StringBuilder.java:274]: These would be better written using `assertEquals()` which provides more information when it fails.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/java.base/share/classes/java/lang/StringBuilder.java [hunk 3]
```diff
@@ -2768,12 +2768,12 @@ public int codePointCount(int start, int end) {
 public int offsetByCodePoints(int start, int codePointCount) {
 	int currentLength = lengthInternal();
 
-	if (start >= 0 && start <= currentLength) {
+	if (start > 0 && start <= currentLength) {
```
[the remark this answers is not in the document]
Comment (mikezhang1234567890) [reply jcl/src/java.base/share/classes/java/lang/StringBuilder.java]: That was a mistake. Fixed in https://github.com/eclipse-openj9/openj9/commit/9ce27382855c5c729d41c845b017212d17e9ed95

**Comment by keithc-ca** [comment]
Jenkins test sanity amac jdk17

**Review by keithc-ca — APPROVED** [review]
